app/controllers/submissions_controller.rb

```diff
@@ -279,20 +279,18 @@ def score
     # To enable hints when scoring a submission, uncomment the next line:
     # send_hints(client_socket, StructuredError.where(submission: @submission))
 
-    # Submit the score
-    if @submission.users.map {|user| lti_outcome_service?(@submission.exercise, user, @submission.study_group_id) }.any?
-      url = finalize_submission_path(@submission)
-      submit_info = transmit_lti_score
-      submit_info&.fetch(:status, [])&.each do |s|
+    # The own score can only be transmitted when the lti_outcome_service is available
+    # if it is not available the user also does not send the score for the programming group members
+    if lti_outcome_service?(@submission.exercise, @current_user, @submission.study_group_id)
+      transmit_lti_score(client_socket)
+    else
+      if @submission.score == @submission.exercise.maximum_score
```

Comment:
This condition is semantically similar to the condition below in the `transmit_lti_score`. Is there a mechanism to avoid the duplication (of the lines below with the `client_socket&.send_data` for `:exercise_finished` in `transmit_lti_score`?